src/Components/Aspire.Azure.Security.KeyVault/AzureKeyVaultKeysComponent.cs

```diff
@@ -0,0 +1,29 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Azure.Security.KeyVault;
+using Azure.Core;
+using Azure.Core.Extensions;
+using Azure.Security.KeyVault.Keys;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace Microsoft.Extensions.Hosting;
+
+internal sealed class AzureKeyVaultKeysComponent : AbstractAzureKeyVaultComponent<KeyClient, KeyClientOptions>
+{
+    protected override IHealthCheck CreateHealthCheck(KeyClient client, AzureSecurityKeyVaultSettings settings)
+        => throw new NotImplementedException();
```

Comment:
Is this going to throw if `settings.DisableHealthChecks == false`?

Reply:
@james-gould I think you need to add this is the two components that don't provide the health-check

```cs
protected override bool GetHealthCheckEnabled(AzureSecurityKeyVaultSettings settings)
{
    return false;
}
```
I will add it myself and approve the PR, unless you think it's wrong and we can then change it.